lib/AST/ASTPrinter.cpp

```diff
@@ -1291,31 +1308,36 @@ void PrintAST::printPattern(const Pattern *pattern) {
 
   case PatternKind::EnumElement: {
     auto elt = cast<EnumElementPattern>(pattern);
-    // FIXME: Print element expr.
+    Printer << "." << elt->getElementDecl()->getBaseName();
     if (elt->hasSubPattern())
       printPattern(elt->getSubPattern());
     break;
   }
 
   case PatternKind::OptionalSome:
     printPattern(cast<OptionalSomePattern>(pattern)->getSubPattern());
-    Printer << '?';
     break;
 
   case PatternKind::Bool:
     Printer << (cast<BoolPattern>(pattern)->getValue() ? tok::kw_true
                                                        : tok::kw_false);
     break;
 
-  case PatternKind::Expr:
-    // FIXME: Print expr.
+  case PatternKind::Expr: {
+    if (Options.PrintExprs) {
+        auto expr = cast<ExprPattern>(pattern)->getSubExpr();
+        visit(expr);
+    }
```

Comment:
Does `Options.PrintExprs` need to be checked here vs letting `visit` handle it?

Reply:
`visitCaseStmt` calls this, which doesn't check for `Options.PrintExprs`, so I thought it was safest to conditionalize it here.

With that said, I removed the condition and all tests pass, so maybe it's safe.

Reply:
Isn't the `visit(expr)` call guaranteed to check it?